src/components/ItemList.svelte

```diff
@@ -20,44 +21,75 @@
 
   export let showFilter = true;
 
+  const features = new Set(
+    flatten(
+      items.map(
+        (item) =>
+          (item.features && item.features.map((f) => f.toLowerCase())) || []
+      )
+    )
+  );
   let filteredItems = items.filter((item) => !isExpired(item.expires));
   let pagedItems;
   let paginated = true;
 
   let currentPage = writable(1);
   setContext("currentPage", currentPage);
 
-  const searchText = getContext("searchText");
-  const goToPage = (page, perPage = DEFAULT_ITEMS_PER_PAGE) => {
-    pagedItems =
-      filteredItems.length > 0
-        ? chunk([...filteredItems], perPage)[page - 1]
-        : [];
-  };
-
-  const showExpired = getContext("showExpired");
-
+  // re-filter items when showExpired or search text changes
   $: {
-    if (paginated) {
-      goToPage($currentPage);
-    } else {
-      goToPage(1, filteredItems.length);
-    }
+    const search = $pageState.search || "";
+    // don't use $currentPage, because we don't want to call this reactively
+    // when paginating
+    currentPage.set(1);
+
+    const featureMatch = (item) =>
+      item.features &&
+      item.features.some((feature) =>
+        feature.toLowerCase().includes(search.toLowerCase())
+      );
+
+    // if the search string is an exact match for a particular feature,
+    // then filter just on that, otherwise filter on search text
+    filteredItems = features.has(search)
+      ? items.filter(
+          (item) =>
+            item.features &&
+            item.features.some(
+              (feature) => feature.toLowerCase() === search.toLowerCase()
+            )
+        )
+      : items.filter(
+          (item) => item.name.includes(search) || featureMatch(item)
```

Comment:
Maybe I'm not following this right, but why do we need `|| featureMatch(item)` when we only need to filter out results that match search text?

Reply:
We also want to match on features, even if only part of their string is matched (e.g. "sea" should match all metrics which have "search" as a feature, even if they do not have search in their name). I can add a comment about this.